### PR TITLE
[WM-1652] move function to utils; handle potentially nested array_type

### PR DESCRIPTION
**Pull request by mspector**

### CODE REVIEW

File: src/libs/utils.js [hunk 1]
```diff
@@ -160,3 +160,16 @@ export const convertValue = _.curry((type, value) => {
...
+  if (_.has('primitive_type', iotype)) {
+    return iotype.primitive_type
+  }
+  if (_.has('optional_type', iotype)) {
+    return `${_.get('optional_type.primitive_type', iotype)} (optional)`
+  }
+  if (_.has('array_type', iotype)) {
+    return `Array[${renderTypeText(_.get('array_type', iotype))}]`
+  }
+  throw new Error(`iotype ${iotype} does not contain a valid type`)
```
Comment (mspector) [inline src/libs/utils.js:174]: I'm not sure that this is the right way to handle the situation where there are no matching types

File: src/libs/utils.test.js [hunk 2]
```diff
@@ -10,3 +10,13 @@ describe('makeCompleteDate', () => {
     expect(makeCompleteDate('2022-01-27T22:27:15.591Z').toString()).toContain('Jan 27, 2022')
   })
 })
+
+describe('submission-common tests', () => {
+  it('variable type text is rendered properly', () => {
+    expect(renderTypeText({ type: 'primitive', primitive_type: 'File' })).toStrictEqual('File')
+    expect(renderTypeText({ type: 'optional', optional_type: { primitive_type: 'String' } })).toStrictEqual('String (optional)')
```
Comment (cjllanwarne) [inline src/libs/utils.test.js:17]:
```suggestion
    expect(renderTypeText({ type: 'optional', optional_type: { type: 'primitive', primitive_type: 'String' } })).toStrictEqual('String (optional)')
```

File: src/libs/utils.test.js [hunk 3]
```diff
@@ -10,3 +10,13 @@ describe('makeCompleteDate', () => {
...
+    expect(renderTypeText({ type: 'array', array_type: { primitive_type: 'Int' } })).toStrictEqual('Array[Int]')
```
Comment (cjllanwarne) [inline src/libs/utils.test.js:18]:
```suggestion
    expect(renderTypeText({ type: 'array', array_type: { type: 'primitive', primitive_type: 'Int' } })).toStrictEqual('Array[Int]')
```

File: src/libs/utils.test.js [hunk 4]
```diff
@@ -10,3 +10,13 @@ describe('makeCompleteDate', () => {
...
+    expect(renderTypeText({ type: 'array', array_type: { optional_type: { primitive_type: 'Int' } } })).toStrictEqual('Array[Int (optional)]')
```
Comment (cjllanwarne) [inline src/libs/utils.test.js:19]:
```suggestion
    expect(renderTypeText({ type: 'array', array_type: { type: 'optional', optional_type: { type: 'primitive', primitive_type: 'Int' } } })).toStrictEqual('Array[Int (optional)]')
```

File: src/libs/utils.test.js [hunk 5]
```diff
@@ -10,3 +10,13 @@ describe('makeCompleteDate', () => {
...
+    expect(renderTypeText({ type: 'array', array_type: { array_type: { array_type: { primitive_type: 'Int' } } } })).toStrictEqual('Array[Array[Array[Int]]]')
```
Comment (cjllanwarne) [inline src/libs/utils.test.js:20]:
```suggestion
    expect(renderTypeText({ type: 'array', array_type: { type: 'array', array_type: { type: 'array', array_type: { type: 'primitive', primitive_type: 'Int' } } } })).toStrictEqual('Array[Array[Array[Int]]]')
```

File: src/libs/utils.js (hunk 1 above)
Comment (cjllanwarne) [reply src/libs/utils.js:174]: So one easy heuristic... assuming we have a bug in this logic, would we prefer the page to crash, or to display some kind of "Unrecognized Type" value and either wait to be told about it, or separately set up some kind of reporting?

**Review by cjllanwarne — APPROVED** [review]
Approved with a few type definition corrections and a thought on error "handling"

File: src/libs/utils.js (hunk 1 above)
Comment (salonishah11) [reply src/libs/utils.js:174]: TOL: Maybe we could say `Type not supported` until we support all types?

**Review by salonishah11 — APPROVED** [review]

**Pull request merged**